### PR TITLE
Introduce the BrowserStack on the CI

### LINKED ISSUE #47: Add current timestamp to project name for builds on BrowserStack
Twin of the https://github.com/ckeditor/ckeditor5-react/issues/44.

### PR DESCRIPTION
**Pull request by ma2ciek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Introduced the BrowserStack on the CI. Closes #47.

---

### Additional information

* [x] Missing badge

### CODE REVIEW

File: src/karma.conf.js [hunk 1]
```diff
@@ -31,7 +31,83 @@ module.exports = function( config ) {
...
+	config.set( karmaConfig );
 };
+
+// Formats name of the build for BrowserStack. It merges a repository name and current timestamp.
+// If env variable `TRAVIS_REPO_SLUG` is not available, the function returns `undefined`.
+//
+// @returns {String|undefined}
+function getBuildName() {
+	const repoSlug = process.env.TRAVIS_REPO_SLUG;
+ 	if ( !repoSlug ) {
```
Comment (pomek) [inline src/karma.conf.js]: A missing blank line before this line. The same in L83 (a blank line after closing a block) and 85 (a blank line before the return statement).

**Comment by pomek** [comment]
What's wrong with the testing CLI from Angular?

```bash
25 10 2018 16:35:12.010:ERROR [reporter]: Can not load reporter "BrowserStack", it is not registered!
  Perhaps you are missing some plugin?
25 10 2018 16:35:16.803:INFO [karma-server]: Karma v3.1.1 server started at http://0.0.0.0:9876/
25 10 2018 16:35:16.804:INFO [launcher]: Launching browsers Firefox, BrowserStack_Edge, BrowserStack_Safari, CHROME_TRAVIS_CI with concurrency unlimited
25 10 2018 16:35:16.808:ERROR [karma-server]: Error: Found 1 load error
    at Server.webServer.listen (/home/travis/build/ckeditor/ckeditor5-angular/node_modules/karma/lib/server.js:180:27)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at Server.emit (events.js:208:7)
    at emitListeningNT (net.js:1387:10)
    at _combinedTickCallback (internal/process/next_tick.js:136:11)
    at process._tickCallback (internal/process/next_tick.js:181:9)
```

I found that in the Travis build. It shouldn't pass.

**Comment by ma2ciek** [comment]
> What's wrong with the testing CLI from Angular?

Not sure. But I don't have an access to exit the process during the `ng test` script execution 😞

File: src/karma.conf.js [hunk 2]
```diff
@@ -31,7 +31,86 @@ module.exports = function( config ) {
...
+
+	if ( shouldEnableBrowserStack() ) {
+		karmaConfig.browserStack = {
+			username: process.env.BROWSER_STACK_USERNAME,
+			accessKey: process.env.BROWSER_STACK_ACCESS_KEY,
+			build: getBuildName(),
+			project: 'ckeditor5'
+		};
+
+		karmaConfig.reporters = [ 'dots', 'BrowserStack' ];
```
[the remark this answers is not in the document]
Comment (ma2ciek) [reply src/karma.conf.js]: Added.

**Comment by ma2ciek** [comment]
Ok, Now it works, but one test on the Edge fails - https://travis-ci.org/ckeditor/ckeditor5-angular/builds/447847018?utm_source=github_status&utm_medium=notification

Edge 17.17134.0 (Windows 10 0.0.0) DemoFormComponent should show form data preview after change FAILED
	Unable to get property 'parent' of undefined or null reference
	TypeError: Unable to get property 'parent' of undefined or null reference...

**Pull request merged**